app/src/main/java/com/samco/trackandgraph/ui/CustomiseCardDialog.kt

```diff
@@ -0,0 +1,170 @@
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Track & Graph.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.samco.trackandgraph.ui
+
+import android.app.Activity
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import com.samco.trackandgraph.database.*
+import com.samco.trackandgraph.database.entity.*
+import com.samco.trackandgraph.databinding.CustomiseCardDialogBinding
+import kotlinx.coroutines.*
+
+const val CUSTOMISE_CARD_DIALOG_FEATURE_KEY = "customise_card_group"
+
+class CustomiseCardDialogFragment : DialogFragment() {
+    private val viewModel by viewModels<CustomiseCardDialogViewModel>()
+    private lateinit var binding: CustomiseCardDialogBinding
+    private val featurePeriodList = listOf(
+        FeatureShowCountPeriod.ALL,
+        FeatureShowCountPeriod.DAILY,
+        FeatureShowCountPeriod.WEEKLY,
+        FeatureShowCountPeriod.MONTHLY,
+        FeatureShowCountPeriod.YEARLY
+    )
+    private val featureMethodList = listOf(
+        FeatureShowCountMethod.COUNT_ENTRIES,
+        FeatureShowCountMethod.SUM_VALUES
+    )
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = CustomiseCardDialogBinding.inflate(inflater, container, false)
+        return activity?.let {
+            val id = requireArguments().getLong(CUSTOMISE_CARD_DIALOG_FEATURE_KEY)
+            viewModel.init(requireActivity(), id)
+            dialog?.setCanceledOnTouchOutside(true)
+
+            listenToViewModelState()
+            binding.root
+        } ?: throw IllegalStateException("Activity cannot be null")
+    }
+
+    private fun getOnPeriodSelectedListener(): AdapterView.OnItemSelectedListener {
+        return object : AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(p0: AdapterView<*>?) {}
+            override fun onItemSelected(p0: AdapterView<*>?, p1: View?, position: Int, p3: Long) {
+                viewModel.setPeriodTo(featurePeriodList[position])
+            }
+        }
+    }
+    private fun getOnMethodSelectedListener(): AdapterView.OnItemSelectedListener {
+        return object : AdapterView.OnItemSelectedListener {
+            override fun onNothingSelected(p0: AdapterView<*>?) {}
+            override fun onItemSelected(p0: AdapterView<*>?, p1: View?, position: Int, p3: Long) {
+                viewModel.setMethodTo(featureMethodList[position])
+            }
+        }
+    }
+    override fun onResume() {
+        super.onResume()
+        dialog?.window?.setLayout(
```

Comment:
I don't think you need this? You can probably remove the scroll view in the dialog layout and just use match_parent/wrap_content everywhere and then you won't need to set this explicitly.